src/core/listener.c

```diff
@@ -88,12 +88,7 @@ MsQuicListenerOpen(
 
 Error:
 
-    if (QUIC_FAILED(Status)) {
-
-        if (Listener != NULL) {
-            CXPLAT_FREE(Listener, QUIC_POOL_LISTENER);
-        }
-    }
+    CXPLAT_DBG_ASSERT(QUIC_SUCCEEDED(Status) || Listener == NULL);
```

Comment:
I was able to remove this check entirely because we can never fail after allocating the `Listener`. But I put an assert here just in case someone changes that in the future.